### ISSUE TITLE
Error raised when validating object against constraints at Class Level

### ISSUE THREAD

**Issue by issamkhadiri1989** [symfony/symfony#59332]
### Symfony version(s) affected

7 and higher

### Description

hello

when trying to add several validation constraints (setup an array of constraints for a given form field) in the Constraints At Class Level, an error is raised as described in the following screenshot :

![image](https://github.com/user-attachments/assets/58b1d443-2d5a-4227-bf37-0fb7c4a22d93)

### How to reproduce

Steps to reproduce the bug:

**Step 1**

create a controller like described in the doc

![image](https://github.com/user-attachments/assets/da860100-d188-4ed2-9c51-0453dcf46085)

**Step 2**

render the form

![image](https://github.com/user-attachments/assets/f4e6d180-2135-47f9-a4c3-2f7242338abe)

**Step 3**

Go to the page /contact

the error is raised

![image](https://github.com/user-attachments/assets/1dd2de45-a34f-443a-9906-affe7e65b385)

to make sure that the problem is coming from the

```
'lastName' => [
    new NotBlank(),
    new Length(['min' => 3]),
],
```
change this configuration by either remove the entire config or just do

```
'lastName' =>new NotBlank(),
```
(use Constraint instead of an array)

and the result is as follows

![image](https://github.com/user-attachments/assets/2cfa0acc-6805-4ed2-adb5-c7208248a6e7)

the form is rendered with no problems

please refer to https://symfony.com/doc/current/form/without_class.html#constraints-at-field-level for more details

thanks in advance,

**Comment by MatTheCat** [comment]
Seems like a documentation bug: https://github.com/symfony/symfony-docs/issues/18969 was wrong since you can only configure `constraints` as a constraint or constraints array. In this case keys would be ignored anyways.

https://github.com/symfony/symfony-docs/pull/19094 probably should be reverted.

**Comment by xabbuh** [comment]
Let's close here in favour of the docs PR: symfony/symfony-docs#20487

**Comment by xabbuh** [comment]
this will be fixed by symfony/symfony-docs#20547
